Feature/firebase data

RojasVSantiago opened the pull request.

RojasVSantiago committed:
New model

Model added: FinishedCocom

RojasVSantiago committed:
Card

Base card for FinishedCocom

RojasVSantiago committed:
Cancel option

Now cocoms can be cancel

RojasVSantiago committed:
formating time

Now time is saved when go to cocoms screen.
Time is set as a String, before saved as a DateTime.
Now route tab show widgets/information based in the cocom.

RojasVSantiago committed:
Finish cocom

Cocom can be finished. this adds the cocom to a list of finished cocoms.

RojasVSantiago committed:
Reports tab

Added report list.
New function, tap a report to see details

RojasVSantiago committed:
Returning Item

When a cocom is finished the informatios is returned to cocoms tab.

RojasVSantiago committed:
Modify Reports

Actualized Finished Cocom String Date (start, end).
Now report start time and end time can be modify by long press on report tab

RojasVSantiago committed:
Backend Posting

When a cocom is finished, send a post to firebase.

RojasVSantiago committed:
Colecting reports

Now reports are loaded from firebase

RojasVSantiago committed:
Patch

Report now can be updated.
Modify a report works in the same way.

SAngelCuadrosV commented on lib/screens/reports/reports_tab.dart:
```diff
@@ -24,66 +23,102 @@ class ReportsTab extends StatefulWidget {
+      });
+    } catch (e) {
+      setState(() {
+        _error = 'Something went wrong. Please try again later.';
```
Estos estados se le muestran al usuario? De ser así, deben ir en español.

RojasVSantiago replied:
Si, se le muestran al usuario en caso de error.

SAngelCuadrosV commented on lib/screens/route/route_tab.dart:
```diff
@@ -12,54 +14,104 @@ import '../../widgets/single_location.dart';
+
+  Widget createButtons() {
+    if (_cocomEnd) {
+      return const Text('Cocom terminada');
```
Cambiar `terminada `por `finalizada`

SAngelCuadrosV commented on lib/widgets/finish_route_button.dart:
```diff
@@ -5,43 +5,86 @@ import '../../widgets/widgets.dart';
+                  content: _startMessage,
+                  actions: [
+                    TextButton(
+                      child: const Text('continuar la cocom'),
```
La primera C en may{uscula

SAngelCuadrosV commented on lib/widgets/modification_modal.dart:
Este texto debe ir en español

SAngelCuadrosV commented on lib/widgets/modification_modal.dart:
Cambiar por "Formato no válido"

SAngelCuadrosV commented on lib/widgets/modification_modal.dart:
`Hora no válida`

SAngelCuadrosV commented on lib/widgets/modification_modal.dart:
`Cancelar`

SAngelCuadrosV commented on lib/widgets/modification_modal.dart:
`Modificar`

RojasVSantiago committed:
Corrections

Text showed to the user was corrected

RojasVSantiago committed:
New Model

InRouteLocation model updated.
Now you can configure hour in single locations.

RojasVSantiago committed:
Report tab

Firebase "folder" for reports changed.

RojasVSantiago committed:
Cocom Express

Express tab changed. Now express tab is a "route tab".
Added a security for cocoms without location added.

SAngelCuadrosV commented on lib/assets/contents/cocoms.dart:
```diff
@@ -58,3 +58,10 @@ Map<String, Cocom> cocomsContent = {
+  id: 'expess',
+  name: 'COCOM EXPRESS',
+  locations: [],
+  information: 'información básica (cambiar luego)',
```
Cambiar esta información a:
`Una Cocom Express es una ruta idealmente a un **punto único** para recoger muestras y ser enviadas con prioridad al centro de análisis. Se puede agregar más de un punto de recogida de ser necesario.`

SAngelCuadrosV commented:
La Ruta Express podría ser un PR adicional.
Para futuros PR: Esto para que los PR sean más fáciles de digerir al momento de ser revisados. Se podría crear una rama desde esta feature si es necesario algún trabajo realizado acá para realizar la Express.

SAngelCuadrosV commented on lib/assets/contents/models/in_route_location.dart:
```diff
@@ -5,15 +5,28 @@ class InRouteLocation {
+        'address': address,
+        'phone': phone,
+        'information': information,
+        'hour': hour
```
Para mejorar la experiencia del usuario sería bueno que a cada punto que ya se haya recogido una muestra, aparezca un feedback como por ejemplo alguno de estos:
- Cambiar el background de color del item en la lista
- Agregar un ícono de Check Verde o Cruz Roja si ha recogido o no la muestra respectivamente.

RojasVSantiago committed:
Hotfix

An error showing every location in report tab was fixed

SAngelCuadrosV commented on lib/screens/reports/reports_tab.dart:
```diff
@@ -53,13 +53,18 @@ class _ReportsTabState extends State<ReportsTab> {
 
+  List<InRouteLocation> _genList(List<InRouteLocation> asd) {
+    final List<InRouteLocation> xd = asd;
+    return xd;
```
Utilizar variables dicientes

RojasVSantiago replied:
esas variables eran una prueba la cual no se usará, las eliminaré.

RojasVSantiago committed:
Route Tab

Added a visual indicator is showing in single locations.
Quantity is reset when a cocom is ended and open again.
Cocom Express information modified.

RojasVSantiago committed:
Variable Change

LocFCocom -> locations:List
